grin/src/Reducer/Base.hs

```diff
@@ -10,22 +10,21 @@ import Grin.Grin
 import Grin.Pretty
 
 -- models cpu registers
-type Env = Map Name RTVal
+type Env v = Map Name (RTVal v)
 
-type SimpleRTVal = RTVal
-data RTVal
-  = RT_ConstTagNode Tag  [SimpleRTVal]
-  | RT_VarTagNode   Name [SimpleRTVal]
+type SimpleRTVal v = RTVal v
+data RTVal v
+  = RT_ConstTagNode Tag  [SimpleRTVal v]
+  | RT_VarTagNode   Name [SimpleRTVal v]
   | RT_ValTag       Tag
   | RT_Unit
-  | RT_Lit          Lit
+  | RT_Lit          v
```

Comment:
What is the motivation behind this generalization?

Reply:
In the long run, the X-GRIN-backends should have their own primitive literals and types. The X-GRIN-backends will need to provide their own way of interpreting, representing them, a way of generate LLVM code etc. Currently the representation and interpretation do coincide with the `Lit` type we use.

In the past, the `Lit` was extended with the String type and its literal, because the need of the Idris-GRIN-Backend, although this type should not be there as it requires special treatment, and the implementation details of such types should be in the X-GRIN-backend codebase.

Reply:
As a mid term goal, I am planning to move all those Idris-GRIN-backend specific codes under the Idris-GRIN codebase.

Reply:
As for the String literals, will they be represented using only primops? The above code only makes the _representation_ pluginable, but not the possible literals. Am I missing something?

Reply:
No. This PR does not contain the abstraction for the literals. This was a necessary almost minimal change for the pluginable PrimOps. I will remove the type parameter from this one.

Reply:
Changes as it started to disturb be too.